### PR TITLE
fix: ignore counts in info tree node [IDE-413]

### PR DESCRIPTION
**Pull request by teodora-sandu**
### Description

Fixes the count of issues and ignored issues in the Tree View so that when the Issue View Options change the counts don't change, only the rendered issues do.

Also adds a missing warning that I should have added as part of https://snyksec.atlassian.net/browse/IDE-332 where there are no ignored issues and only showing ignored issues or there are no open issues and only showing open issues.

I enabled the default linter in IntellJ and it formatted a file. Let me know if you'd like me to undo that.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs
Screenshot 2024-06-18 at 09 31 49

Screenshot 2024-06-18 at 09 32 07

### CODE REVIEW

File: src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowSnykScanListenerLS.kt [hunk 1]
```diff
@@ -163,11 +169,20 @@ class SnykToolWindowSnykScanListenerLS(
             if (filterTree) {
                 addInfoTreeNodes(rootNode, snykResults.values.flatten().distinct(), fixableIssuesCount)
 
+                val includeIgnoredIssues =
```
Comment (teodora-sandu) [inline src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowSnykScanListenerLS.kt:172]: Moved the filtering of issues to here. That way we don't affect the counts of issues and ignored issues based on the settings.

File: src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowSnykScanListenerLS.kt [hunk 2]
```diff
@@ -242,11 +257,27 @@ class SnykToolWindowSnykScanListenerLS(
                 )
             }
         }
+
```
Comment (teodora-sandu) [inline src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowSnykScanListenerLS.kt:260]: Added the warnings.

File: src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt [hunk 3]
```diff
@@ -212,17 +216,7 @@ class SnykLanguageClient : LanguageClient, Disposable {
         check(snykScan.product == "code" || snykScan.product == "oss") { "Expected Snyk Code or Snyk OSS scan result" }
         if (snykScan.issues.isNullOrEmpty()) return emptyMap()
 
-        val pluginSettings = pluginSettings()
```
Comment (teodora-sandu) [inline src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt:215]: Removed the filtering from here.

File: src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowSnykScanListenerLS.kt [hunk 4]
```diff
@@ -163,11 +169,20 @@ class SnykToolWindowSnykScanListenerLS(
             if (filterTree) {
                 addInfoTreeNodes(rootNode, snykResults.values.flatten().distinct(), fixableIssuesCount)
 
+                val includeIgnoredIssues =
+                    !(settings.isGlobalIgnoresFeatureEnabled && !settings.ignoredIssuesEnabled)
+                val includeOpenedIssues =
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowSnykScanListenerLS.kt:174]: This double negation is kinda hard to grasp.

File: src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowSnykScanListenerLS.kt (hunk 2 above)
Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowSnykScanListenerLS.kt:260]: Ah, very nice.

File: src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowSnykScanListenerLS.kt [hunk 5]
```diff
@@ -242,11 +257,27 @@ class SnykToolWindowSnykScanListenerLS(
...
+            rootNode.add(
+                InfoTreeNode(
+                    "Adjust your Issue View Options to see ignored issues.",
+                    project,
+                ),
+            )
+        } else if (ignoredIssuesCount == 0 && !settings.openIssuesEnabled) {
+            rootNode.add(
+                InfoTreeNode(
+                    "Adjust your Issue View Options to open ignored issues.",
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowSnykScanListenerLS.kt:271]: "open ignored issues"?

**Review by bastiandoetsch — APPROVED** [review]

**Review by bastiandoetsch — APPROVED** [review]
LGTM, just some minor nitpicks. I'd make that boolean logic easier to understand, but not blocking.

**Pull request merged**